source/set-mutable.d.ts

```diff
@@ -0,0 +1,37 @@
+import {Except} from './except';
+import {Mutable} from './mutable';
+
+/**
+Create a type that converts the given keys from `readonly` to mutable. The remaining keys are kept as is.
+
+Use-case: You want to define a single model where the only thing that changes is whether or not some of the keys are mutable.
+
+@see [`Mutable`](source/mutable.d.ts) - a type that sets all (not selectively) keys as mutable.
+
+@example
+```
+import {SetMutable} from 'type-fest';
+
+type Foo = {
+	readonly a: number;
+	b: readonly string[];
+	readonly c: boolean;
+}
+
+type SomeMutable = SetMutable<Foo, 'b' | 'c'>;
+// type SomeMutable = {
+// 	readonly a: number;
+// 	b: readonly string[]; // Was already mutable and still is, type of the property remains unaffected.
+// 	c: boolean; // Is now mutable.
+// }
+```
+*/
+export type SetMutable<BaseType, Keys extends keyof BaseType> =
```

Comment:
I'm kind of wondering why we simply doesn't add `Keys extends keyof BaseType = keyof BaseType` to `Mutable<>` and makes this implementation be the future `Mutable<>` where one can optionally pick a subset of keys to make mutable.

Does it make sense to have two so similar ones? I know @mmkal had this same wondering in https://github.com/sindresorhus/type-fest/pull/157#discussion_r531389560 but it was resolved by making all `Set*` require a keys attribute without any discussion on why this one doesn't simply replace `Mutable<>`, I would like to discuss the latter 🙂

Reply:
I think it's done to be consistent with the other `Set-` types and their non-set siblings.

Reply:
Yeah, I guess @sindresorhus can explain 🙂

Reply:
I think the initial reason was consistency. For example, `SetOptional`. We couldn't extend the built-in `Partial`, right?

Thinking more about it now and seeing how the "duplication" continues, I'm also thinking that we should merge them.

Can you think of any downsides?

Reply:
Honestly the `Set-` types are kinda confusing in naming. There's no way to immediately know how `SetRequired` differs from `Required` until you check the docs.

And since there's no `Mutable<T>` built-in type, merging those two is a no-brainer. We just have to document them properly to make it obvious that there's a `Mutable<T, keys?>` option.

Reply:
I merged these two types. Take a gander. We can roll back that commit with necessary.

Reply:
Continued in https://github.com/sindresorhus/type-fest/issues/171